src/overlays/actors/ovl_En_Wonder_Talk2/z_en_wonder_talk2.c

```diff
@@ -26,19 +31,274 @@ const ActorInit En_Wonder_Talk2_InitVars = {
+    player = PLAYER;
+    this->unk_158++;
+    if (this->switchFlag >= 0 && Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        if (this->unk_15A == 0) {
+            this->actor.flags &= -2;
+            this->unk_15A = 1;
+            return;
+        }
+
+    } else {
+        if (func_8002F194(&this->actor, globalCtx) != 0) {
+            if ((this->switchFlag >= 0) && (this->unk_154 != 2)) {
+                Flags_SetSwitch(globalCtx, this->switchFlag);
+                osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ セーブしたよ！おもいっきり！ %x\n" VT_RST, this->switchFlag);
+            }
+
+            this->actionFunc = func_80B3A10C;
+            return;
+        }
+
+        yawDiffTemp = this->actor.yawTowardsLink - this->actor.posRot.rot.y;
+        yawDiff = ABS(yawDiffTemp);
+
+        if ((!((40.0f + this->unk_15C) < (*actor).xzDistFromLink)) &&
+            ((!(fabsf(player->actor.posRot.pos.y - (*actor).posRot.pos.y) > 100.0f)) && (yawDiff < 0x4000))) {
+            if (this->unk_158 >= 2) {
+                osSyncPrintf("\n\n");
+                // Transparent Message Kimi Set
+                osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 透明メッセージ君せっと %x\n" VT_RST, this->actor.params);
+                // Save Information
+                osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ セーブ情報 \t           %x\n" VT_RST, this->switchFlag);
+                // Specified message type
+                osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 指定メッセージ種類     %x\n" VT_RST, this->unk_150);
+                // Actual message type
+                osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 実質メッセージ種類     %x\n" VT_RST, this->actor.textId);
+                // Specified range
+                osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 指定範囲               %d\n" VT_RST, this->actor.posRot.rot.z);
+                // Processing range
+                osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 処理範囲               %f\n" VT_RST, this->unk_15C);
+                switch (this->unk_154) {
+                    case 0:
+                        // Normal
+                        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ 通常 ☆☆ \n" VT_RST);
+                        break;
+                    case 2:
+                        // Check only
+                        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ チェックのみ ☆☆ \n" VT_RST);
+                        break;
+                    case 3:
+                        // Lock only
+                        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆ ロックのみ ☆☆ \n" VT_RST);
+                        break;
+                }
+            }
+
+            this->unk_158 = 0;
+            func_8002F1C4(&this->actor, globalCtx, this->unk_15C + 50.0f, 100.0f, 0);
+        }
+    }
+}
+
+void func_80B3A3D4(EnWonderTalk2* this, GlobalContext* globalCtx) {
+    u8 dialogState;
+
+    if (BREG(2)) {
+        // Oh
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ わー %d\n" VT_RST, func_8010BDBC(&globalCtx->msgCtx));
+    }
+    dialogState = func_8010BDBC(&globalCtx->msgCtx);
+
+    switch (dialogState) {
```

Comment:
Is the temp actually needed here?
```suggestion

    switch (func_8010BDBC(&globalCtx->msgCtx)) {
```